jenkins/jenkinspipeline-set-up-hnsesb-job.txt

```diff
@@ -0,0 +1,190 @@
+// This pipeline is created to build the application using build config and then deploy using deployment config
+pipeline {
+	
+	parameters {
+		choice(name: 'ACTION', choices: ['SETUP', 'DELETE'], description: 'Required action. If selected DELETE, this job will delete all references of Application in selected namespace')
+		
+		choice(name: 'NAMESPACE', choices: ['Dev', 'Test'], description: 'Project namespace in openshift where you want to create this application')
+		
+        string(name: 'BRANCH_NAME', defaultValue: '3946-jenkins-job-for-new-hnsesb-insntance', description: 'Name of the branch')
+
+        text(name: 'Application', defaultValue: 'hnsesb-sample', description: 'Enter the name of application. This name will be used to define the URL. If application with same name already exist, this job will delete that application')
+
+		string(name: 'PHARMANET_USER', defaultValue: 'hniconnectsvc', description: 'Pharmanet user for pharmanet web service')
+		
+		password(name: 'PHARMANET_PASSWORD', defaultValue: '', description: 'Passowrd for pharmanet web service')
```

Comment:
Did we talk about storing user/passwords in Credentials? I believe you could then change the parameters to be credentialsId and then use withCredentials to extract the user/pass when you need them. Probably not required at this stage but is a nice, secure way to handle user/passwords.

Reply:
Correct, we decided to add that. I will update the job.